receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -226,36 +226,57 @@ func parseJobAndInstance(dest pcommon.Map, job, instance string) {
+	for j := 0; j < rm.ScopeMetrics().Len(); j++ {
+		scope := rm.ScopeMetrics().At(j)
+		if scopeName == scope.Scope().Name() && scopeVersion == scope.Scope().Version() {
+			metric := scope.Metrics().AppendEmpty()
+			sum := metric.SetEmptySum()
+			sum.SetIsMonotonic(true)
+			sum.SetAggregationTemporality(pmetric.AggregationTemporalityCumulative)
+			metric.SetName(ls.Get(labels.MetricName))
+			addDatapoints(sum.DataPoints(), ls, ts)
+			return
+		}
+	}
+
+	scope := rm.ScopeMetrics().AppendEmpty()
+	scope.Scope().SetName(scopeName)
+	scope.Scope().SetVersion(scopeVersion)
+	metric := scope.Metrics().AppendEmpty()
+	sum := metric.SetEmptySum()
+	// Set monotonic to true since Prometheus counters are always monotonic
+	sum.SetIsMonotonic(true)
+	sum.SetAggregationTemporality(pmetric.AggregationTemporalityCumulative)
+	metric.SetName(ls.Get(labels.MetricName))
+	addDatapoints(sum.DataPoints(), ls, ts)
 }
 
 func addGaugeDatapoints(rm pmetric.ResourceMetrics, ls labels.Labels, ts writev2.TimeSeries) {
-	// TODO: Cache metric name+type+unit and look up cache before creating new empty metric.
-	// In OTel name+type+unit is the unique identifier of a metric and we should not create
-	// a new metric if it already exists.
-
 	scopeName := ls.Get("otel_scope_name")
 	scopeVersion := ls.Get("otel_scope_version")
-	// TODO: If the scope version or scope name is empty, get the information from the collector build tags.
-	// More: https://opentelemetry.io/docs/specs/otel/compatibility/prometheus_and_openmetrics/#:~:text=Metrics%20which%20do%20not%20have%20an%20otel_scope_name%20or%20otel_scope_version%20label%20MUST%20be%20assigned%20an%20instrumentation%20scope%20identifying%20the%20entity%20performing%20the%20translation%20from%20Prometheus%20to%20OpenTelemetry%20(e.g.%20the%20collector%E2%80%99s%20prometheus%20receiver)
 
-	// Check if the name and version present in the labels are already present in the ResourceMetrics.
-	// If it is not present, we should create a new ScopeMetrics.
-	// Otherwise, we should append to the existing ScopeMetrics.
+	// Check if the name and version present in the labels are already present in the ResourceMetrics
 	for j := 0; j < rm.ScopeMetrics().Len(); j++ {
 		scope := rm.ScopeMetrics().At(j)
 		if scopeName == scope.Scope().Name() && scopeVersion == scope.Scope().Version() {
-			addDatapoints(scope.Metrics().AppendEmpty().SetEmptyGauge().DataPoints(), ls, ts)
+			metric := scope.Metrics().AppendEmpty()
+			metric.SetName(ls.Get(labels.MetricName))
+			addDatapoints(metric.SetEmptyGauge().DataPoints(), ls, ts)
 			return
 		}
 	}
 
 	scope := rm.ScopeMetrics().AppendEmpty()
 	scope.Scope().SetName(scopeName)
 	scope.Scope().SetVersion(scopeVersion)
-	m := scope.Metrics().AppendEmpty().SetEmptyGauge()
-	addDatapoints(m.DataPoints(), ls, ts)
+	metric := scope.Metrics().AppendEmpty()
+	metric.SetName(ls.Get(labels.MetricName))
+	addDatapoints(metric.SetEmptyGauge().DataPoints(), ls, ts)
```

Comment:
This PR focuses on counters; I'd recommend that we don't make changes to the gauge functions here. If things need to change for consistency, let's do it in a separate PR :)